core/envelope-sim/src/test/java/fr/sncf/osrd/envelope_sim/EnvelopeSimPathTest.java

```diff
@@ -37,81 +41,84 @@ void findHighGradePosition() {
+        modes.put(Range.closed(0.0, 10.0), new NonElectrified());
+        modes.put(Range.closed(3.0, 7.0), new Electrified("1500"));
+        modes.put(Range.closed(7.1, 10.0), new Electrified("25000"));
+        var path = new EnvelopeSimPath(10, new double[] { 0, 10 }, new double[] { 0 }, ImmutableRangeMap.copyOf(modes),
+                new HashMap<>());
+        var modeAndProfileMap = path.getElectrificationMap(null, null, null, true);
+
+        assertTrue(fullyCovers(modeAndProfileMap, 10));
+
+        assertEquals(modeAndProfileMap.get(0.), new NonElectrified());
+        assertEquals(modeAndProfileMap.get(4.), new Electrified("1500"));
+        assertEquals(modeAndProfileMap.get(7.05), new NonElectrified());
+        assertEquals(modeAndProfileMap.get(7.2), new Electrified("25000"));
     }
 
     @ParameterizedTest
     @ValueSource(booleans = {true, false})
     void getCatenaryModeAndProfile(boolean withEmptyPowerRestrictionMap) {
         var path = EnvelopeSimPathBuilder.withElectricalProfiles1500();
 
-        RangeMap<Double, EnvelopeSimPath.ElectrificationConditions> modeAndProfileMap;
+        RangeMap<Double, Electrification> modeAndProfileMap;
         if (withEmptyPowerRestrictionMap)
-            modeAndProfileMap = path.getElecCondMap("2", ImmutableRangeMap.of(), Map.of("Restrict1", "1"));
+            modeAndProfileMap = path.getElectrificationMap("2", ImmutableRangeMap.of(), Map.of("Restrict1", "1"));
         else
-            modeAndProfileMap = path.getElecCondMap("2", null, Map.of("Restrict1", "1"));
+            modeAndProfileMap = path.getElectrificationMap("2", null, Map.of("Restrict1", "1"));
 
-        assertEquals(7, modeAndProfileMap.asMapOfRanges().size());
+        assertTrue(fullyCovers(modeAndProfileMap, path.length));
 
-        assertEquals(modeAndProfileMap.get(2.0), new EnvelopeSimPath.ElectrificationConditions("1500", null, null));
-        assertEquals(modeAndProfileMap.get(3.5), new EnvelopeSimPath.ElectrificationConditions("1500", "A", null));
-        assertEquals(modeAndProfileMap.get(5.5), new EnvelopeSimPath.ElectrificationConditions("1500", "C", null));
-        assertEquals(modeAndProfileMap.get(6.5), new EnvelopeSimPath.ElectrificationConditions("1500", "B", null));
+        assertEquals(9, modeAndProfileMap.asMapOfRanges().size());
+
+        assertEquals(modeAndProfileMap.get(2.0), new Electrified("1500", null, null));
+        assertEquals(modeAndProfileMap.get(3.5), new Electrified("1500", "A", null));
+        assertEquals(modeAndProfileMap.get(5.5), new Electrified("1500", "C", null));
+        assertEquals(modeAndProfileMap.get(6.5), new Electrified("1500", "B", null));
     }
 
     @Test
     void getCatenaryModeAndProfileWithPowerRestrictions() {
         var path = EnvelopeSimPathBuilder.withElectricalProfiles1500();
 
-        RangeMap<Double, EnvelopeSimPath.ElectrificationConditions> modeAndProfileMap;
+        RangeMap<Double, Electrification> modeAndProfileMap;
```

Comment:
Needed ?

Reply:
I thought you asked if the change was needed at first, your message isn't very explicit 
Indeed the variable declaration isn't needed :grin: